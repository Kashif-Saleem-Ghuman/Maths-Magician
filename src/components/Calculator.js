import React from 'react';

class Calculator extends React.PureComponent {
  render() {
    return (
      <>
        <article className="container">
          <section className="result">
            <span>0</span>
          </section>
          <section className="wrapper">
            <button type="button">AC</button>
            <button type="button">+/-</button>
            <button type="button">%</button>
            <button type="button" className="operators">{'\u00F7'}</button>
            <button type="button">7</button>
            <button type="button">8</button>
            <button type="button">9</button>
            <button type="button" className="operators">x</button>
            <button type="button">4</button>
            <button type="button">5</button>
            <button type="button">6</button>
            <button type="button" className="operators">-</button>
            <button type="button">1</button>
            <button type="button">2</button>
            <button type="button">3</button>
            <button type="button" className="operators">+</button>
            <button type="button" className="zero">0</button>
            <button type="button" className="dot">.</button>
            <button type="button" className="operators">=</button>
          </section>
        </article>

      </>

    );
  }
}

export default Calculator;
