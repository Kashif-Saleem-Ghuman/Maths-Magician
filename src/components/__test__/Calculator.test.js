import { render } from '@testing-library/react';
import Calculator from '../Calculator';

describe('Testing Calculator', () => {
  test('should render correctly', () => {
    render(<Calculator />);
  });
  test('should have a length of 1', () => {
    const { container } = render(<Calculator />);
    expect(container.querySelectorAll('.calc-head-wrapper')).toHaveLength(1);
  });
});
