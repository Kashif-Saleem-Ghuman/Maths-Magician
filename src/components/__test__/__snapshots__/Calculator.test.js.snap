// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  <section
    className="calc-head-wrapper"
  >
    <div>
      <p
        className="calc-p"
      >
        Lets do some math!
      </p>
    </div>
    <article
      className="container"
    >
      <section
        className="result"
      >
        <span>
          0
           
           
        </span>
      </section>
      <section
        className="wrapper"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="dot"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </section>
    </article>
  </section>,
  <footer>
    <span
      className="footer-text"
    >
      Created by Kashif Saleem
    </span>
    <a
      className="github"
      href="https://github.com/Kashif-Saleem-Ghuman"
      rel="noreferrer"
      target="_blank"
    >
      Github
    </a>
  </footer>,
]
`;

exports[`matches sn 1`] = `
Array [
  <section
    className="calc-head-wrapper"
  >
    <div>
      <p
        className="calc-p"
      >
        Lets do some math!
      </p>
    </div>
    <article
      className="container"
    >
      <section
        className="result"
      >
        <span>
          0
           
           
        </span>
      </section>
      <section
        className="wrapper"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="dot"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </section>
    </article>
  </section>,
  <footer>
    <span
      className="footer-text"
    >
      Created by Kashif Saleem
    </span>
    <a
      className="github"
      href="https://github.com/Kashif-Saleem-Ghuman"
      rel="noreferrer"
      target="_blank"
    >
      Github
    </a>
  </footer>,
]
`;

exports[`matches snapshot 1`] = `
Array [
  <section
    className="calc-head-wrapper"
  >
    <div>
      <p
        className="calc-p"
      >
        Lets do some math!
      </p>
    </div>
    <article
      className="container"
    >
      <section
        className="result"
      >
        <span>
          0
           
           
        </span>
      </section>
      <section
        className="wrapper"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="dot"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="operators"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </section>
    </article>
  </section>,
  <footer>
    <span
      className="footer-text"
    >
      Created by Kashif Saleem
    </span>
    <a
      className="github"
      href="https://github.com/Kashif-Saleem-Ghuman"
      rel="noreferrer"
      target="_blank"
    >
      Github
    </a>
  </footer>,
]
`;

exports[`renders when the home is updated 1`] = `
Array [
  <main>
    <h2>
      Welcome to Our page!
    </h2>
    <p
      className="paragraph"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptas suscipit deserunt commodi iusto quasi voluptatem incidunt modi perspiciatis ipsam dolorum at ut pariatur sunt veniam saepe, eveniet quis consequuntur fuga assumenda magni amet cum Iusto enim fugiat provident aliquam nobis laborum aperiam similique aspernatur obcaecati sunt asperiores quidem totam numquam at hic fuga temporibus Distinctio commodi nostrum laudantium accusantium quam.
    </p>
    <p
      className="paragraph2"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptas suscipit deserunt commodi iusto quasi voluptatem incidunt modi perspiciatis ipsam dolorum at ut pariatur sunt veniam saepe, eveniet quis consequuntur fuga assumenda magni amet cum Iusto enim fugiat provident aliquam nobis laborum aperiam similique aspernatur obcaecati sunt.
    </p>
  </main>,
  <footer>
    <span
      className="footer-text"
    >
      Created by Kashif Saleem
    </span>
    <a
      className="github"
      href="https://github.com/Kashif-Saleem-Ghuman"
      rel="noreferrer"
      target="_blank"
    >
      Github
    </a>
  </footer>,
]
`;
